plugins/modules/setup.ps1

```diff
@@ -3,505 +3,1143 @@
-        ansible_architecture = $win32_os.OSArchitecture
-        ansible_domain = $domain_suffix
-        ansible_fqdn = $fqdn
-        ansible_hostname = $ip_props.HostName
-        ansible_netbios_name = $win32_cs.Name
-        ansible_kernel = $osversion.Version.ToString()
-        ansible_nodename = $fqdn
-        ansible_machine_id = Get-MachineSid
-        ansible_owner_contact = ([string] $win32_cs.PrimaryOwnerContact)
-        ansible_owner_name = ([string] $win32_cs.PrimaryOwnerName)
-        # FUTURE: should this live in its own subset?
-        ansible_reboot_pending = $ansible_reboot_pending
-        ansible_system = $osversion.Platform.ToString()
-        ansible_system_description = ([string] $win32_os.Description)
-        ansible_system_vendor = $win32_cs.Manufacturer
-    }
-}
+            // 0 == ComputerNameNetBIOS
+            NativeMethods.GetComputerNameExW(0, buffer, ref size);
+            buffer.EnsureCapacity(size);
 
-if($gather_subset.Contains('powershell_version')) {
-    $ansible_facts += @{
-        ansible_powershell_version = ($PSVersionTable.PSVersion.Major)
+            if (!NativeMethods.GetComputerNameExW(0, buffer, ref size))
+                throw new Win32Exception("Failed to get ComputerName");
+            NetBIOSName = buffer.ToString();
+        }
     }
 }
+'@
+
+$factMeta = @(
+    @{
+        Subsets = 'all_ipv4_addresses', 'all_ipv6_addresses'
+        Code = {
+            $interfaces = [System.Net.NetworkInformation.NetworkInterface]::GetAllNetworkInterfaces()
+            $ips = @(foreach($interface in $interfaces) {
+                # Win32_NetworkAdapterConfiguration did not return the local IPs so we replicate that here.
+                $interface.GetIPProperties().UnicastAddresses | Where-Object {
+                    $_.Address.ToString() -notin @('::1', '127.0.0.1')
+                } | ForEach-Object -Process {
+                    $_.Address.ToString()
+                }
+            })
+
+            $ansibleFacts.ansible_ip_addresses = $ips
+        }
+    },
+    @{
+        Subsets = 'bios'
+        Code = {
+            $bios = New-Object -TypeName Ansible.Windows.Setup.SMBIOSInfo
+
+            $releaseDate = if ($bios.ReleaseDate) {
+                $bios.ReleaseDate.ToUniversalTime().ToString('MM/dd/yyyy')
+            }
+            $ansibleFacts.ansible_bios_date = $releaseDate
+            $ansibleFacts.ansible_bios_version = $bios.SMBIOSBIOSVersion
+            $ansibleFacts.ansible_product_name = $bios.Model
+            $ansibleFacts.ansible_product_serial = $bios.SerialNumber
+        }
+    },
+    @{
+        Subsets = 'date_time'
+        Code = {
+            $datetime = (Get-Date)
+            $datetimeUtc = $datetime.ToUniversalTime()
+
+            $ansibleFacts.ansible_date_time = @{
+                date = $datetime.ToString("yyyy-MM-dd")
+                day = $datetime.ToString("dd")
+                epoch = (Get-Date ($datetimeUtc) -UFormat '+%s')
```

Comment:
Discussed in comments, but recommend `epoch_local` (or `local_epoch`, or whatever name) as a way to get at the old value easily.

Reply:
Went with `epoch_local` in this case.